### ISSUE TITLE
`symfony/console` locked to v3 with php 8

### ISSUE THREAD

**Issue by giraz82** [cakephp/cakephp#16109]
### Description

I created a new CakePHP project on PHP 8.0.12.
I found out that `symfony/console` is locked at `v3.4.47`.
This prevent me to require other packages, such as php-cs-fixer (but also others):
```
 - friendsofphp/php-cs-fixer[v3.2.0, ..., v3.2.1] require symfony/console ^4.4.20 || ^5.1.3
 ```
I also tried with `-W` flag with no luck.
After I downgraded to PHP 7.4 and ran a composer update, I got `symfony/console` `v5.3.10` and I solved other following problems as well.

I think it's not correct to have `symfony/console` locked to a very old version on PHP 8, since the problem doesn't occurs on PHP 7.4.

Thanx

### CakePHP Version

4.3.1

### PHP Version

8.0.12

**Comment by ADmad** [comment]
`cakephp/repl` and `cakephp/migrations` are the two packages that have `symfony/console` as indirect dependency. But both [phinx](https://github.com/cakephp/phinx/blob/e71f2dea5706d2abdfecc56623eefa19a78eca7e/composer.json#L31) and [psysh](https://github.com/bobthecow/psysh/blob/01281336c4ae557fe4a994544f30d3a1bc204375/composer.json#L19) allow v4 and higher for `symfony/console`.

Did you try reinstalling after deleting your `composer.lock` file?

**Comment by ADmad** [comment]
Creating a new project using `composer/app` does clamp `symfony/console` to `v3.4.47`. Removing `cakephp/repl` and running `composer update -W` does bump it up to `v5.3.10`, so `psy/psysh` is the culprit.

**Comment by ADmad** [comment]
```
composer why symfony/console -t
symfony/console v3.4.47 Symfony Console Component
├──composer/composer 2.1.12 (requires symfony/console ^2.8.52 || ^3.4.35 || ^4.4 || ^5.0 || ^6.0)
│  └──cakephp/debug_kit 4.5.0 (requires composer/composer ^1.3 | ^2.0)
│     └──cakephp/app (requires (for development) cakephp/debug_kit ^4.5)
├──psy/psysh v0.10.9 (requires symfony/console ~5.0|~4.0|~3.0|^2.4.2|~2.3.10)
│  └──cakephp/repl 0.1.0 (requires psy/psysh @stable)
│     └──cakephp/app (requires (for development) cakephp/repl ^0.1)
├──robmorgan/phinx 0.12.9 (requires symfony/console ^3.4|^4.0|^5.0)
│  └──cakephp/migrations 3.3.0 (requires robmorgan/phinx ^0.12)
│     └──cakephp/app (requires cakephp/migrations ^3.2)
└──symfony/var-dumper v4.4.33 (conflicts symfony/console <3.4) (circular dependency aborted here)
```

**Comment by giraz82** [comment]
Thank you @ADmad.
I removed `cakephp/rep` but both running `composer update -W` or deleting `composer.lock` and running a new `composer install` I still get:
```
composer why symfony/console -t
symfony/console v3.4.47 Symfony Console Component
├──composer/composer 2.1.12 (requires symfony/console ^2.8.52 || ^3.4.35 || ^4.4 || ^5.0 || ^6.0)
│  └──cakephp/debug_kit 4.5.0 (requires composer/composer ^1.3 | ^2.0)
│     └──cakephp/app dev-master (requires (for development) cakephp/debug_kit ^4.5)
└──robmorgan/phinx 0.12.9 (requires symfony/console ^3.4|^4.0|^5.0)
   └──cakephp/migrations 3.3.0 (requires robmorgan/phinx ^0.12)
      └──cakephp/app dev-master (requires cakephp/migrations ^3.2)
```
I also tried `composer require symfony/console:^5.3` or `composer require symfony/console:^5.3 -W` but no luck as well.

Then I removed also `cakephp/debug_kit` and `cakephp/migrations`, so composer removed also `symfony/console`.
After that I tried to install again `symfony/console`, hoping to get v5.3:
```
composer require symfony/console

Using version ^5.3 for symfony/console
./composer.json has been updated
Running composer update symfony/console
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - symfony/console[v5.3.0, ..., v5.3.10] require symfony/service-contracts ^1.1|^2 -> satisfiable by symfony/service-contracts[v1.1.1, ..., v1.1.9, v2.0.0, ..., v2.4.0].
    - symfony/service-contracts[v1.1.1, ..., v1.1.8] require php ^7.1.3 -> your php version (8.0.11) does not satisfy that requirement.
    - symfony/service-contracts[v1.1.9, v2.1.2, ..., v2.2.0] require psr/container ^1.0 -> found psr/container[1.0.0, 1.1.0, 1.1.1, 1.1.2] but the package is fixed to 2.0.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - symfony/service-contracts v2.0.0 requires php ^7.2.9 -> your php version (8.0.11) does not satisfy that requirement.
    - symfony/service-contracts v2.0.1 requires php ^7.2.5 -> your php version (8.0.11) does not satisfy that requirement.
    - symfony/service-contracts v2.4.0 requires psr/container ^1.1 -> found psr/container[1.1.0, 1.1.1, 1.1.2] but the package is fixed to 2.0.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires symfony/console ^5.3 -> satisfiable by symfony/console[v5.3.0, ..., v5.3.10].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
```
Then:
```
composer require symfony/console -W
Using version ^5.3 for symfony/console
./composer.json has been updated
Running composer update symfony/console --with-all-dependencies
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install symfony/console v5.3.2 (conflict analysis result)
    - Conclusion: don't install symfony/console v5.3.4 (conflict analysis result)
    - Conclusion: don't install symfony/console v5.3.6 (conflict analysis result)
    - Conclusion: don't install symfony/console v5.3.7 (conflict analysis result)
    - Conclusion: don't install symfony/console v5.3.10 (conflict analysis result)
    - cakephp/cakephp 4.3.1 requires league/container ^4.1.1 -> satisfiable by league/container[4.1.2].
    - symfony/service-contracts[v1.1.1, ..., v1.1.8] require php ^7.1.3 -> your php version (8.0.11) does not satisfy that requirement.
    - symfony/service-contracts v2.0.0 requires php ^7.2.9 -> your php version (8.0.11) does not satisfy that requirement.
    - symfony/service-contracts v2.0.1 requires php ^7.2.5 -> your php version (8.0.11) does not satisfy that requirement.
    - Root composer.json requires symfony/console ^5.3 -> satisfiable by symfony/console[v5.3.0, ..., v5.3.10].
    - Conclusion: don't install psr/container 2.0.2 (conflict analysis result)
    - Conclusion: don't install psr/container 1.1.2 (conflict analysis result)
    - symfony/console v5.3.0 requires symfony/service-contracts ^1.1|^2 -> satisfiable by symfony/service-contracts[v1.1.1, ..., v1.1.9, v2.0.0, ..., v2.4.0].
    - Conclusion: don't install symfony/service-contracts v2.1.2 (conflict analysis result)
    - Conclusion: don't install symfony/service-contracts v2.1.3 (conflict analysis result)
    - Conclusion: don't install symfony/service-contracts v2.2.0 (conflict analysis result)
    - Conclusion: don't install symfony/service-contracts v2.4.0 (conflict analysis result)
    - cakephp/cakephp is locked to version 4.3.1 and an update of this package was not requested.
    - symfony/service-contracts v1.1.9 requires psr/container ^1.0 -> satisfiable by psr/container[1.0.0, 1.1.0, 1.1.1, 1.1.2].
    - You can only install one version of a package, so only one of these can be installed: psr/container[1.0.0, 1.1.0, 1.1.1, 1.1.2, 2.0.0, 2.0.1, 2.0.2].
    - league/container 4.1.2 requires psr/container ^2.0.0 -> satisfiable by psr/container[2.0.0, 2.0.1, 2.0.2].
    - Conclusion: don't install psr/container 2.0.1 (conflict analysis result)
```

So I'm blocked here. 😓

**Comment by othercorey** [comment]
If you are blocked by this, can you create an app using 4.2?

**Comment by ADmad** [comment]
@giraz82 Please run following sequence of commands and provide the output:
```
composer create-project cakephp/app myapp
cd myapp
composer remove cakephp/repl
composer update -W
composer why symfony/console -t
```

**Comment by garas** [comment]
There is conflict on `psr/container`:

`cakephp/cakephp 4.3.1` requires `psr/container ^2.0`.

`symfony/console v5.3.2` requires `symfony/service-contracts: ^1.1|^2` which requires `psr/container ^1.0`.

If you need `symfony/console`, you could try to install `symfony/console v6.0.0-BETA1`

**Comment by giraz82** [comment]
@ADmad Unfortunately removing `cakephp/repl` didn't unlock `symfony/console`.
```
composer create-project cakephp/app cakephp4.3
cd cakephp4.3
composer remove cakephp/repl
composer update -W
composer why symfony/console -t

symfony/console v3.4.47 Symfony Console Component
├──composer/composer 2.1.12 (requires symfony/console ^2.8.52 || ^3.4.35 || ^4.4 || ^5.0 || ^6.0)
│  └──cakephp/debug_kit 4.5.0 (requires composer/composer ^1.3 | ^2.0)
│     └──cakephp/app (requires (for development) cakephp/debug_kit ^4.5)
└──robmorgan/phinx 0.12.9 (requires symfony/console ^3.4|^4.0|^5.0)
   └──cakephp/migrations 3.3.0 (requires robmorgan/phinx ^0.12)
      └──cakephp/app (requires cakephp/migrations ^3.2)
```

 @othercorey Everything fine on 4.2 👍🏻 
```
composer create-project cakephp/app cakephp4.2 "4.2.*"
cd cakephp4.2
composer why symfony/console -t

symfony/console v5.3.10 Eases the creation of beautiful and testable command line interfaces
├──composer/composer 2.1.12 (requires symfony/console ^2.8.52 || ^3.4.35 || ^4.4 || ^5.0 || ^6.0)
│  └──cakephp/debug_kit 4.4.4 (requires composer/composer ^1.3 | ^2.0)
│     └──cakephp/app (requires (for development) cakephp/debug_kit ^4.4)
├──psy/psysh v0.10.9 (requires symfony/console ~5.0|~4.0|~3.0|^2.4.2|~2.3.10)
│  └──cakephp/app (requires (for development) psy/psysh @stable)
├──robmorgan/phinx 0.12.9 (requires symfony/console ^3.4|^4.0|^5.0)
│  └──cakephp/migrations 3.1.0 (requires robmorgan/phinx ^0.12)
│     └──cakephp/app (requires cakephp/migrations ^3.0)
└──symfony/var-dumper v5.3.10 (conflicts symfony/console <4.4) (circular dependency aborted here)
```
@garas you beated me on time 😅 
I just came at same conclusion.

`cakephp/cakephp v4.3.1` requires `psr/container^2.0`, and this doesn't permit to install `symfony/service-contracts ^2.4`, required by `symfony/console v5.3.2`

**Comment by giraz82** [comment]
Anyway my initial goal was to install
```
friendsofphp/php-cs-fixer
phpmd/phpmd
povils/phpmnd
```
And `symfony/console v6.0.0-BETA1` doesn't help on this 😞 
The fun thing is that the problem doesn't occur on php 7.4.

**Comment by ADmad** [comment]
I think up coming symfony 5.4 also has psr/container `^1.0 || ^2.0`.

BTW you can just use `phpcbf` which comes with `phpcs` instead of friendsofphp/php-cs-fixer :)

**Comment by ADmad** [comment]
@giraz82 Please leave a comment or 👍 this PR https://github.com/thephpleague/container/pull/234.

**Comment by giraz82** [comment]
@ADmad are you sure that https://github.com/thephpleague/container/pull/234 is the only problem?
`cakephp/cakephp` has `psr/container: ^2.0` ad direct dependency.
I'm not super expert with dependencies so I could be wrong.

**Comment by ADmad** [comment]
> cakephp/cakephp has psr/container: ^2.0 ad direct dependency.

Yes, that's to match the constraint used by league/container. Once it's updated we too can update the constraint.

**Comment by othercorey** [comment]
league/container 4.2 allows psr/container 1.1. We've updated our constraints to match and should be fixed in cakephp 4.3.1.
